packages/wrangler/src/pages/dev.ts

```diff
@@ -476,15 +521,80 @@ export const Handler = async (args: PagesDevArguments) => {
 					routesModule,
 					defineNavigatorUserAgent,
 				});
+
+				for (const dep in bundle.dependencies) {
+					const resolvedDep = resolve(functionsDirectory, dep);
+					/*
+					 * EXCLUDE:
+					 *   - the "/functions" directory because we're already watching it
+					 *   - everything in "./.wrangler", as it's mostly cache and
+					 *     temporary files
+					 *   - any bundle dependencies we are already watching
+					 *   - anything outside of the current working directory, since we
+					 *     are expecting `wrangler pages dev` to be run from the Pages
+					 *     project root folder
+					 */
+					if (
```

Comment:
How does this deal with things like packages in node_modules? Should we exclude those too?

Reply:
I thought abt that, but decided not to exclude them. My main concern was hitting perf bottleneck due to watching too many files. I have no feel whatsoever of how many deps a project has on average. If we are worried abt perf hits or other things, we can def exclude. FWICT, watching `node_modules` is not an expectation users have?